File: src/1262-Greatest-Sum-Divisible-by-Three/1262.java
class Solution {
    public int maxSumDivThree(int[] nums) {
        int[] res = new int[3];
        for (int a : nums) {
            int[] tmp = res.clone();
            for (int i : tmp) {
                res[(i + a)%3] = Math.max(res[(i + a)%3], i + a);
            }
        }
        return res[0];
    }
}

File: src/1262-Greatest-Sum-Divisible-by-Three/1262.py
class Solution:
    def maxSumDivThree(self, nums: List[int]) -> int:
        dp = [0, 0, 0]
        for a in nums:
            for i in dp[:]:
                dp[(i + a) % 3] = max(dp[(i + a) % 3], i + a)
        return dp[0]